framework/src/modules/chain/bft/finality_manager.js

```diff
@@ -304,22 +304,39 @@ class FinalityManager extends EventEmitter {
 			return true;
 		}
 
+		// Order the two block headers such that earlierBlock must be forged first
+		let earlierBlock = delegateLastBlock;
+		let laterBlock = blockHeader;
 		if (
-			delegateLastBlock.maxHeightPreviouslyForged ===
-				blockHeader.maxHeightPreviouslyForged &&
-			delegateLastBlock.height >= blockHeader.height
+			earlierBlock.maxHeightPreviouslyForged >
+				laterBlock.maxHeightPreviouslyForged ||
+			(earlierBlock.maxHeightPreviouslyForged ===
+				laterBlock.maxHeightPreviouslyForged &&
+				earlierBlock.maxHeightPrevoted > laterBlock.maxHeightPrevoted) ||
+			(earlierBlock.maxHeightPreviouslyForged ===
+				laterBlock.maxHeightPreviouslyForged &&
+				earlierBlock.maxHeightPrevoted === laterBlock.maxHeightPrevoted &&
+				earlierBlock.height > laterBlock.height)
+		) {
+			earlierBlock = blockHeader;
+			laterBlock = earlierBlock;
```

Comment:
Should not this be:
```
laterBlock = delegateLastBlock;
```
?
It looks like here `earlierBlock` and `laterBlock` are going to be the same.